central/buildbot.py

```diff
@@ -22,18 +22,18 @@ def make_netstring(s):
     return str(len(s)).encode('ascii') + b':' + s + b','
 
 
-def make_build_request(repo, pr_id, job_id, baserev, headrev, patch, who,
+def make_build_request(repo, pr_id, job_id, baserev, headrev, who,
                        comment):
     """Creates a build request binary blob in the format expected by the
     buildbot."""
 
     request_dict = {
-        'branch': '',
+        'branch': 'refs/pull/%d/merge' % pr_id,
```

Comment:
Same question as before, are we sure we want this?
The "merge" branch is updated asynchronously and we might be building the wrong change (rarely though). A rebuild would fix it, but it's something we have to be aware of.